src/unstract/adapters/embedding/embedding_adapter.py

```diff
@@ -25,6 +25,10 @@ def get_name() -> str:
     def get_description() -> str:
         return ""
 
+    @staticmethod
+    def get_provider() -> str:
```

Comment:
NIT: We can use `@abstractmethod` decorator to indicate that all subclasses should implement this method to ensure we don't forget it in any existing / future adapters.
Another possible way is to raise a `NotImplementedError` but this might catch it only at run-time

Reply:
Yes.  I have the same comment too as @chandrasekharan-zipstack. [Test connection](https://github.com/Zipstack/unstract-adapters/blob/782e04907e2c41fbf6f5f088f6a39cc9a0a9bb62/src/unstract/adapters/base.py#L43) follows this approach, however at the adapters level. Same can be done for embedding and llm.

Reply:
@chandrasekharan-zipstack @gaya3-zipstack Is this what you are suggesting?

```
@staticmethod
@abstractmethod
def get_provider() -> str:
    raise NotImplementedError("Subclasses must implement get_provider() method")
```

Won't this raise error on runtime? I don't need that behavior because if something goes wrong, the token information itself will not be recorded in the usage table then.

Reply:
I tried this @Deepak-Kesavan 
```
@staticmethod
@abstractmethod
def mock_method() -> str:
    pass
```
Now if you don't define this function in its subclasses, we obtain this error
![image](https://github.com/user-attachments/assets/3177bb57-7fac-4f8f-a7eb-c17a42fdb3b6)

Here's the traceback. Check if this might help what we're trying to do
[traceback.txt](https://github.com/user-attachments/files/16302591/traceback.txt)

Also when we add a new adapter and something goes wrong with this error - it'll likely be caught during development or testing in staging so I don't think it'll be a huge concern of tokens not being recorded

Reply:
Made it as abstract method.